samza-core/src/test/scala/org/apache/samza/util/TestCoordinatorStreamUtil.scala

```diff
@@ -84,4 +84,19 @@ class TestCoordinatorStreamUtil {
     Assert.assertEquals(configFromCoordinatorStream.get(keyForEmptyVal), emptyVal)
     Assert.assertFalse(configFromCoordinatorStream.containsKey(keyForNullVal))
   }
+
+  @Test
+  def testWriteConfigToCoordinatorStream() {
+    val addConfig = new util.HashMap[String, String]
+    addConfig.put("job.name", "test-job-name")
+    addConfig.put("job.id", "i001")
+    addConfig.put("job.coordinator.system", "samzatest")
+    addConfig.put("systems.samzatest.test","test")
+    addConfig.put("test.only","nothing")
+    addConfig.put("systems.samzatest.samza.factory", "org.apache.samza.system.MockSystemFactory")
+    val config = new MapConfig(addConfig)
+    val configMap = CoordinatorStreamUtil.buildCoordinatorStreamConfig(config)
+
+    CoordinatorStreamUtil.writeConfigToCoordinatorStream(configMap)
```

Comment:
Can we verify if the configs are piped to the mock system producer? 
Now that you are adding tests, maybe add one for resetJobConfig to make sure the configs not part of the old configs get deleted?

Reply:
As discussed, in order for us to mock and verify, we need to mock multiple nested data, it is not worth the effort to do it for now, as the code are simply copied from existing code.